### PR TITLE
Workaround GTC-2986 by checking if version is successfully created after get 5XX response from API

### PR DESCRIPTION
**Pull request by jterry64**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [X] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [X] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [X] Check the commit's or even all commits' message styles matches our requested structure.
- [X] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?
We often see during nightly jobs to create new GLAD/integrated alerts tables a 500 response from the API on version creation, but the version often actually does get created. This then gives a 400 error on retry that the version already exists, and causes us to error out before finishing the sync job, even though the table was actually successfully uploaded.

## What is the new behavior?
On getting error from creating new version, check if version did successfully get created before returning an exception.

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: src/datapump/clients/data_api.py [hunk 1]
```diff
@@ -129,16 +129,26 @@ def create_vector_version(
...
+            except DataApiResponseError:
+                raise e
 
     def create_aux_asset(
         self, dataset: str, version: str, payload: Dict[str, Any]
     ) -> Dict[str, Any]:
-
         uri = f"{GLOBALS.data_api_uri}/dataset/{dataset}/{version}/assets"
-        return self._send_request(ValidMethods.post, uri, payload)["data"]
+        return self.get_version(ValidMethods.post, uri, payload)["data"]
```
Comment (danscales) [inline src/datapump/clients/data_api.py:151]: Was there a bad merge here?  I don't see any reason to change create_aux_asset() to use get_version(), and it is the wrong set of args for get_version.

**Review by danscales — APPROVED** [review]
Thanks for doing this!  Looks good, except for possibly extraneous change to create_aux_asset()

File: src/datapump/clients/data_api.py (hunk 1 above)
Comment (jterry64) [reply src/datapump/clients/data_api.py:151]: Oh possibly, I had to backwards merge some hotfixes from master to develop that may have gotten mixed up

Comment (jterry64) [reply src/datapump/clients/data_api.py]: removed

**Pull request merged**